src/option_menu.c

```diff
@@ -0,0 +1,929 @@
+                "\tbl PutWindowTilemap\n"
+                "\tmovs r0, 0x1\n"
+                "\tmovs r1, 0x3\n"
+                "\tbl CopyWindowToVram\n"
+                "\tadd sp, 0x30\n"
+                "\tpop {r3}\n"
+                "\tmov r8, r3\n"
+                "\tpop {r4-r7}\n"
+                "\tpop {r0}\n"
+                "\tbx r0\n"
+                "\t.align 2, 0\n"
+                "_08088AF8: .4byte gText_FrameType\n"
+                "_08088AFC: .4byte sOptionsMenuPtr\n");
+}
+#endif
+
+
+//sub_8088B00
+static void sub_8088B00(u8 taskId)
+{
+    gFieldCallback = sub_807DF64;
+    SetMainCallback2(gMain.savedCallback);
+    FreeAllWindowBuffers();
+    
+    gSaveBlock2Ptr->optionsTextSpeed = (u8) sOptionsMenuPtr->option[MENUITEM_TEXTSPEED];
+    gSaveBlock2Ptr->optionsBattleSceneOff = (u8) sOptionsMenuPtr->option[MENUITEM_BATTLESCENE];
+    gSaveBlock2Ptr->optionsBattleStyle = (u8) sOptionsMenuPtr->option[MENUITEM_BATTLESTYLE];
+    gSaveBlock2Ptr->optionsSound = sOptionsMenuPtr->option[MENUITEM_SOUND];
+    gSaveBlock2Ptr->optionsButtonMode = (u8) sOptionsMenuPtr->option[MENUITEM_BUTTONMODE];
+    gSaveBlock2Ptr->optionsWindowFrameType = (u8) sOptionsMenuPtr->option[MENUITEM_FRAMETYPE];
+    SetPokemonCryStereo(gSaveBlock2Ptr->optionsSound);
+    Free(sOptionsMenuPtr);
+    sOptionsMenuPtr = NULL;
+    DestroyTask(taskId);
+}
+
+    
+    
+//sub_8088BD0
+static void sub_8088BD0(void)
+{
+    FillWindowPixelBuffer(0, 0x11);
+    AddTextPrinterParameterized(WIN_TEXT_OPTION, 2, gText_MenuOptionOption, 8, 1, TEXT_SPEED_FF, NULL);
+    PutWindowTilemap(0);
+    CopyWindowToVram(0, 3);
+}
+
+
+//sub_8088C0C
+static void sub_8088C0C(void)
+{
+    u8 h;
+    h = 2;
+    
+    FillBgTilemapBufferRect(1, 0x1B3, 1, 2, 1, 1, 3);
+    FillBgTilemapBufferRect(1, 0x1B4, 2, 2, 0x1B, 1, 3);
+    FillBgTilemapBufferRect(1, 0x1B5, 0x1C, 2, 1, 1, 3);
+    FillBgTilemapBufferRect(1, 0x1B6, 1, 3, 1, h, 3);
+    FillBgTilemapBufferRect(1, 0x1B8, 0x1C, 3, 1, h, 3);
+    FillBgTilemapBufferRect(1, 0x1B9, 1, 5, 1, 1, 3);
+    FillBgTilemapBufferRect(1, 0x1BA, 2, 5, 0x1B, 1, 3);
+    FillBgTilemapBufferRect(1, 0x1BB, 0x1C, 5, 1, 1, 3);
+    FillBgTilemapBufferRect(1, 0x1AA, 1, 6, 1, 1, h);
+    FillBgTilemapBufferRect(1, 0x1AB, 2, 6, 0x1A, 1, h);
+    FillBgTilemapBufferRect(1, 0x1AC, 0x1C, 6, 1, 1, h);
+    FillBgTilemapBufferRect(1, 0x1AD, 1, 7, 1, 0x10, h);
+    FillBgTilemapBufferRect(1, 0x1AF, 0x1C, 7, 1, 0x10, h);
+    FillBgTilemapBufferRect(1, 0x1B0, 1, 0x13, 1, 1, h);
+    FillBgTilemapBufferRect(1, 0x1B1, 2, 0x13, 0x1A, 1, h);
+    FillBgTilemapBufferRect(1, 0x1B2, 0x1C, 0x13, 1, 1, h);
+    CopyBgTilemapBufferToVram(1);
+}
+
+//sub_8088D8C
+static void sub_8088D8C(void)
+{
+    u8 i;
+    
+    FillWindowPixelBuffer(1, 0x11);
+    for (i = 0; i < MENUITEM_COUNT; i++)
+    {
+        AddTextPrinterParameterized(1, 2, sOptionMenuItemsNames[i], 8, (u8) ((i*(GetFontAttribute(2, FONTATTR_MAX_LETTER_HEIGHT))) + 2) - i, 0xFF, 0);    
```

Comment:
1) space on left and right of `*`.
2) 0 -> NULL
3) why did you use TEXT_SPEED_FF above but not here?